src/gafaelfawr/models/link.py

```diff
@@ -0,0 +1,41 @@
+"""Representation for a ``Link`` HTTP header."""
+
+from __future__ import annotations
+
+import re
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+
+@dataclass
+class LinkData:
+    """Holds the data returned in an RFC 8288 ``Link`` header."""
+
+    prev_url: Optional[str]
+    """The URL of the previous page, or `None` for the first page."""
+
+    next_url: Optional[str]
+    """The URL of the next page, or `None` for the last page."""
+
+    first_url: Optional[str]
+    """The URL of the first page."""
+
+    @classmethod
+    def from_header(cls, header: Optional[str]) -> LinkData:
+        """Parse an RFC 8288 ``Link`` with pagination URLs."""
+        links = {}
+        if header:
+            elements = header.split(",")
+            for element in elements:
+                match = re.match(' *<([^>]+)>; rel="([^"]+)"', element)
```

Comment:
I see that you're basically building a hash with "prev", "next", and "first" and their targets, but it took a little doing.  Maybe name your match groups with something like "type" and "target" ?